packages/captp/src/captp.js

```diff
@@ -206,21 +254,45 @@ export const makeCapTP = (
     // associated slot number.
     const slot = valToSlot.get(val);
     assert.typeof(slot, 'string');
+
     return slot;
   }
 
+  /**
+   * @type {ConvertSlotToVal<CapTPSlot>}
+   */
+  const assertValIsLocal = val => {
+    const slot = valToSlot.get(val);
+    assert(
+      !(slot && slot[1] === '-'),
+      X`Value ${val} slot ${slot} indicates it is remote; we are expecting only local`,
+    );
+  };
+
+  const { serialize: assertOnlyLocal } = makeMarshal(assertValIsLocal);
+  const isOnlyLocal = specimen => {
+    // Try marshalling the object, but throw on references to remote objects.
+    try {
+      assertOnlyLocal(specimen);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  };
```

Comment:
OMG I hate to see a try/catch used to convert an assert into a predicate, where predicate failure is considered non-exceptional. I use your `isFoo/assertFoo/checkFoo` pattern alot and love it! Is there anything we can or should do here that is not too much trouble for the short term?